FindMyClass/app/api/googleCalendar.js

```
@@ -0,0 +1,45 @@
+import { getAuth } from "firebase/auth";
+import AsyncStorage from '@react-native-async-storage/async-storage'; // ✅ Retrieve stored access token
+
+const fetchGoogleCalendarEvents = async () => {
+  try {
+    const auth = getAuth();
+    if (!auth.currentUser) {
+      console.error("User not logged in. Please sign in with Google.");
+      return [];
+    }
+    // ✅ Get the stored Google access token
+    const googleAccessToken = await AsyncStorage.getItem("googleAccessToken");
+
+    if (!googleAccessToken) {
+      console.error("No Google OAuth access token found. Please sign in again.");
+      return [];
+    }
+
+    // ✅ Make the request with the correct OAuth access token
+    const response = await fetch(
+      `https://www.googleapis.com/calendar/v3/calendars/primary/events?maxResults=10&orderBy=startTime&singleEvents=true`,
+      {
+        method: "GET",
+        headers: {
+          Authorization: `Bearer ${googleAccessToken}`, // ✅ Correct Google OAuth token
+          Accept: "application/json",
+        },
+      }
+    );
+
+    const data = await response.json();
+    if (data.error) {
+      console.error("Google API Error:", data.error);
+      return [];
+    }
```

Instead of just logging the error and returning an empty array, should we return an error object or throw an exception for better debugging?